src/logger.test.js

```diff
@@ -1,101 +1,265 @@
-/* eslint-disable */
 import * as logger from './logger';
 import * as utils from './utils';
 import { TracerGlobals } from './globals';
-
-jest.spyOn(global.console, 'log');
-global.console.log.mockImplementation(() => {});
+import { ConsoleWritesForTesting } from '../testUtils/consoleMocker';
 
 describe('logger', () => {
-  const spies = {};
-  const oldEnv = Object.assign({}, process.env);
-  spies.log = jest.spyOn(logger, 'log');
+  const times = (x, callback) => {
+    for (let i = 0; i < x; i++) {
+      callback();
+    }
+  };
+
+  test('info -> simple flow', () => {
+    utils.setDebug();
+    TracerGlobals.setTracerInputs({});
+
+    logger.info('Test');
+
+    expect(ConsoleWritesForTesting.getLogs()).toEqual([
+      {
+        msg: '#LUMIGO# - INFO - "Test"',
+        obj: undefined,
+      },
+    ]);
+  });
+
+  test('info -> with object', () => {
+    utils.setDebug();
+    TracerGlobals.setTracerInputs({});
+
+    logger.info('Test', { a: 2 });
+
+    expect(ConsoleWritesForTesting.getLogs()).toEqual([
+      {
+        msg: '#LUMIGO# - INFO - "Test"',
+        obj: '{"a":2}',
+      },
+    ]);
+  });
+
+  test('info -> debug is off', () => {
+    TracerGlobals.setTracerInputs({});
+
+    logger.info('Test');
+
+    expect(ConsoleWritesForTesting.getLogs()).toEqual([]);
+  });
+
+  test('info -> store logs on -> not printing', () => {
+    utils.setDebug();
+    utils.setStoreLogsOn();
+    TracerGlobals.setTracerInputs({});
+
+    logger.info('Test');
+
+    expect(ConsoleWritesForTesting.getLogs()).toEqual([]);
+  });
+
+  test('info -> store logs on -> printing after 50 dups', () => {
+    utils.setDebug();
```

Comment:
Are you sure we need to `setDebug` for this test?
`debug` isn't related to `storeLogs`, isn't it?
Same for other similar tests.

Reply:
Yes we can remove this, it doesn't matter